Added new potion

JQChong opened the pull request:
I added a new potion, which is to increase the user's size by 50% and reduce the amount of damage received by the user by 25% for 30 seconds.

JQChong committed:
Add files via upload

JQChong committed:
Delete PotionCommonEffects.java

JQChong committed:
Delete EnlargePotionBottle.png

JQChong committed:
Delete EnlargePotion.prefab

JQChong committed:
Add files via upload

JQChong committed:
Add files via upload

JQChong committed:
Add files via upload

Cervator commented:
Hey @JQChong ! Thanks for the PR :-) Sorry for taking a little while to get to it!

Unfortunately it looks like you were using the web UI to add/edit files, which is really awkward and tends to make the history really messy. For instance you managed to entirely delete then re-add `PotionCommonEffects` making the whole file show up as changed instead of just the line you added.

You've also got tab characters in `EnlargePotion.prefab` when we use spaces instead, and whitespace may be a little off in general as well.

Have you got a working source code workspace set up in IntelliJ? There are ways to both get syntax/whitespace checking, code validation, testing and so on, plus it makes pushing changes to GitHub way easier and cleaner.

Your potion itself looks fine, but the formatting/file issues are problematic, and the increase size effect isn't actually connected to any logic that'll make it happen. You can see how effects work using the AlterationEffects module, and add a size/height changing effect to it (the `MovementDebugCommands` class has a `playerHeight` command). But first you'd probably be better off making sure you can run the game from source and push changes to GitHub for review.

We can help but it is way easier to do live support in our IRC channel than here on GitHub or even in the GCI task system (the notifications from there are really easy to miss)

JQChong committed:
Formatting

Editted the formatting of the EnlargePotion.prefab and
PotionCommonEffects.java file

Cervator commented:
Hey again @JQChong - thanks for the fixes but the INCREASE_SIZE still isn't hooked up to anything. It is just declared, but there's no code to actually change the player's size :-)

We do want grow/shrink potions and there are commands for changing the player's height that could be used to make a new effect that'll cause that. Not super easy though and it *really* needs a fully working source workspace to investigate the related code.

There is a task for setting up to run from source that would probably be idea to try first, and you can still reuse the changes from here.

Cervator commented:
Am now realizing this is probably linked with https://github.com/Terasology/AlterationEffects/pull/7 and they need to be merged together as part of https://codein.withgoogle.com/dashboard/task-instances/6602914749480960

No time for me tonight but pinging @xrtariq2594 :-)
